HIP/hip-0000-add-software-version-to-event.md

```diff
@@ -0,0 +1,111 @@
+---
+hip: <HIP number (this is determined by the HIP editor)>
+title: Add Software Version to Events
+author: Edward Wertz <@edward-swirldslabs>
+working-group: a list of the technical and business stakeholders' name(s) and/or username(s), or name(s) and email(s).
+type: Standards Track
+category: Core
+needs-council-approval: <Yes | No>
+status: Draft
+created: 4/19/2023
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/720
+updated: <comma separated list of dates>
+requires: <HIP number(s)>
+replaces: <HIP number(s)>
+superseded-by: <HIP number(s)>
+---
+
+## Abstract
+
+Add the software version to events that are serialized and to the event interface. This permits the application to
+determine which version of software was used to create each event, and optionally to ignore events created by previous
+versions.
+
+## Motivation
+
+This change should come before the implementation of the preconsensus event stream. The preconsensus event stream
+will save events to disk prior to their reaching consensus. On restart of the application all events in the preconsensus
+event stream will be replayed to reach consensus. If the restart coincides with a software upgrade, the new software
+version will receive events created from the previous software version. This change would enable the application to
+decide whether to act on the transactions generated by the older software version.
```

Comment:
```suggestion
This change is a prerequisite for a planned platform feature called the preconsensus event stream. The preconsensus event stream writes events to disk prior to the events reaching consensus. When a node is
restarted, events from this stream will be replayed, starting with events that came after the state snapshot
loaded into the node, and ending with the last events that were saved prior to the node shutting down.

When the system performs an upgrade, this new system of replaying events will make it possible for events
created before the software upgrade to reach consensus after the software upgrade. (Currently, these events
are simply discarded). The change proposed by this HIP will enable the application to decide whether to act on the transactions generated by the older software version, or if those transactions should be ignored.
```

Reply:
Michael moved the hip to 721 which prevents sign off and commit of the suggestion.  Suggestion accepted and adopted in the next commit.